source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -271,6 +272,50 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       response_headers_to_add_ = std::move(response->response_headers_to_add);
     }
 
+    absl::optional<Http::Utility::QueryParams> modified_query_parameters;
+    if (!response->query_parameters_to_set.empty()) {
+      modified_query_parameters =
+          Http::Utility::parseQueryString(request_headers_->Path()->value().getStringView());
+      ENVOY_STREAM_LOG(
+          trace, "ext_authz filter set query parameter(s) on the request:", *decoder_callbacks_);
+      for (const auto& [key, value] : response->query_parameters_to_set) {
+        ENVOY_STREAM_LOG(trace, "'{}={}'", *decoder_callbacks_, key, value);
+        // TODO(esmet): Sanitize key/value and/or declare the security posture that we trust the
+        // authorization server.
```

Comment:
Trusting the auth server seems obvious but I still need to circle back to this TODO

Reply:
File an issue will be better I think?